components/eamxx/src/control/atmosphere_driver.cpp

```diff
@@ -283,6 +283,9 @@ void AtmosphereDriver::create_grids()
     setup_iop ();
   }
 
+  // The ATM processes (potentially) need output parameter file in set_grids()
```

Comment:
I'm not sure we should handle it this way. If I understand correctly, you want atm procs to decide what to do with, say, diagnostic fields, depending on whether they are needed in I/O. I would have the AD do the heavy lifting. In particular:

- atm procs create field requests as "Diagnostic"
- if IO needs a diag field, then:
  - AD creates the field and puts it in the FM
  - AD sets it in the atm proc
  - atm proc sees the field during atmbuf setup as well as initialize
- if IO does not need the diag field, then:
  - AD never creates it
  - atm procs realize the field is not there when atmbuf setup or initialize are called, and acts accordingly
  
In particular, during the atmbuf setup method, the atm proc should:
- request mem for the diag field if it was never set by the AD
- NOT request mem for the diag field if the AD set it

Reply:
The reason for this is that I want the infrastructure to handle this as much as possible. If we set the param list in the atm proc, the physics parametrizations developers need to learn how to navigate the output params. I think it's much easier for them to do something like
```c++
void MyAP::requested_buffer_size_in_bytes() {
  int nb = 0;
  if (not has_diagnostic_field("blah")) {
    // AD did not create the blah field. Evidently, it's not needed in I/O,
    // so we need to handle it internally, as a scratch view
    nb += <the size of blah>;
  }
  return nb;
}

void MyAP::init_buffers_ (...) {
  if (not has_diagnostic_field("blah")) {
    // AD did not create the blah field. Evidently, it's not needed in I/O,
    // so we need to handle it internally, as a scratch view
    m_blah = view_2d<Real>(mem,m_ncols,m_nlevs);
  }
}

void MyAP::initialize_impl(...) {
  if (has_diagnostic_field("blah")) {
    // AD did create the blah field. Evidently, it IS needed in I/O,
    // so we need to use its view in our calculations
    m_blah = get_field_out("blah").get_view<Real**>();
  }
}
```

Reply:
Yeah, that's a good point about navigating a parameter list. Let me think about this and I'll push a different concept.